### PR TITLE
Direct users to `cargo concordium test --out ..` instead `cargo test`

### PR DESCRIPTION
**Pull request by Bargsteen**
## Purpose

This is to reduce the risk of running tests against an outdated wasm module.
The --out flag, and other build options, were added to the `cargo concordium test` command in the most recent cargo concordium version.

**Should not be published before we've published the new version of cargo concordium.**

## Changes

- Refer users to `cargo concordium test --out ...` instead of `cargo test`.
- Update various texts to fit in with the new command
- Simplify the high-level view of integration testing

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: source/mainnet/smart-contracts/guides/integration-test-contract.rst [hunk 1]
```diff
@@ -29,20 +29,23 @@ The high-level process of adding integration tests to your existing smart contra
...
+      fn my_test() {
+        ...
+        let module = module_load_v1("concordium-out/module.wasm.v1").unwrap();
+        ...
+      }
 
-   where you specify the path the Wasm file built in step two (details will be explained later).
-4. Run your tests with ``cargo test``
-5. If your tests fail due to a bug in your smart contract, fix the bug and recompile your contract with ``cargo-concordium`` as in the second step before running the tests again. Otherwise, the changes are not reflected in the tests.
+   where you specify the path to the Wasm file built in next step.
```
Comment (dg-concordium) [inline source/mainnet/smart-contracts/guides/integration-test-contract.rst:46]:
```suggestion
   where you specify the path to the Wasm file built in the next step.
```

File: source/mainnet/smart-contracts/guides/integration-test-contract.rst [hunk 2]
```diff
@@ -108,8 +111,10 @@ Deploying smart contract modules is a two-step process.
 First, you load the module with the function |module_load_v1|_, then you deploy it to the chain with the method |Chain_module_deploy_v1|_.
 Loading as a separate step allows you to reuse the loaded module across multiple tests for efficiency.
 
-The module to load should be a ``wasm`` module compiled with ``cargo concordium build``.
-For example, for ``cargo concordium build --embed-schema --out my_module.wasm.v1``, you write:
+The module to load should be a ``wasm`` module compiled with ``cargo concordium build`` or ``cargo concordium test --out path/to/wasm/module``.
```
Comment (dg-concordium) [inline source/mainnet/smart-contracts/guides/integration-test-contract.rst:114]: _Should_ it be wasm or _must_ it be wasm? I don't like should because it indicates room for doubt.

File: source/mainnet/smart-contracts/tutorials/piggy-bank/testing.rst [hunk 3]
```diff
@@ -237,6 +237,16 @@ Then compile your contract with:
 This produces the Wasm module ``piggy_bank_part2.wasm.v1`` in the root of your project, unless you have any typos or bugs in your code.
 If that is the case, try to fix them using the helpful error messages from the compiler or go back to the end of part 1 and copy the full contract code again.
 
+.. note::
+
+   Since the tests run against the compiled Wasm module, there is a risk of accidentally using an outdated Wasm module.
+   To circumvent this, cargo concordium's test command can both *build* and test you contract, which ensures that you always test the newest version of your code.
```
Comment (dg-concordium) [inline source/mainnet/smart-contracts/tutorials/piggy-bank/testing.rst:243]:
```suggestion
   To circumvent this, cargo concordium's test command can both build *and* test your contract, which ensures that you always test the newest version of your code.
```

File: source/mainnet/smart-contracts/tutorials/piggy-bank/testing.rst [hunk 4]
```diff
@@ -237,6 +237,16 @@ Then compile your contract with:
...
+   To do so, use the ``--out`` parameter testing:
```
Comment (dg-concordium) [inline source/mainnet/smart-contracts/tutorials/piggy-bank/testing.rst:244]:
```suggestion
   To do so, use the ``--out`` parameter testing:
```
```suggestion
   To do so, use the ``--out`` parameter when testing:
```

File: source/mainnet/smart-contracts/tutorials/piggy-bank/testing.rst [hunk 5]
```diff
@@ -400,7 +410,9 @@ Run the test to check that it compiles and succeeds.
 
 .. code-block:: console
 
-   $cargo test
+   $cargo concordium test --out piggy_bank_part2.wasm.v1
+
+Note that we are using the ``--out`` flag for ``cargo concordium`` to ensure that the module is freshly compiled and thus matches the newest version of your smart contract code.
```
Comment (dg-concordium) [inline source/mainnet/smart-contracts/tutorials/piggy-bank/testing.rst:415]:
```suggestion
Note that the ``--out`` flag for ``cargo concordium`` ensures that the module is freshly compiled and thus matches the newest version of your smart contract code.
```

**Review by dg-concordium — APPROVED** [review]

File: source/mainnet/smart-contracts/guides/integration-test-contract.rst [hunk 6]
```diff
@@ -108,8 +111,10 @@ Deploying smart contract modules is a two-step process.
...
+Using the test command is ideal, as that will both compile the module *and* run the tests.
+By compiling the module every time, you ensure that the tests run on the newest version of your code.
+For example, for ``cargo concordium test --embed-schema --out my_module.wasm.v1``, you write:
```
Comment (abizjak) [inline source/mainnet/smart-contracts/guides/integration-test-contract.rst:117]: I think we need to say somewhere that they need cargo-concordium version 2.9.0 at least for this to work.

File: source/mainnet/smart-contracts/tutorials/piggy-bank/testing.rst (hunk 3 above)
Comment (abizjak) [inline source/mainnet/smart-contracts/tutorials/piggy-bank/testing.rst:243]: Perhaps we could add here that if you use `cargo concordium test --out` then you 
- build the module and output it
- and run tests on it.

And thus you should deploy that module, which means you are deploying a module you have tested, exactly that one.

**Review by abizjak — APPROVED** [review]
Looks good.

Cargo-concordium 2.9 is released now, so this can be deployed as soon as the comments are resolved.

File: source/mainnet/smart-contracts/tutorials/piggy-bank/testing.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (Bargsteen) [reply source/mainnet/smart-contracts/tutorials/piggy-bank/testing.rst:415]: Right, I've tried to rephrase it to make it more clear.

**Review by limemloh — APPROVED** [review]

File: source/mainnet/smart-contracts/tutorials/piggy-bank/testing.rst (hunk 3 above)
Comment (abizjak) [reply source/mainnet/smart-contracts/tutorials/piggy-bank/testing.rst]: Yeah, I think that's good.

**Comment by abizjak** [comment]
@dg-concordium Can we get this deployed tomorrow? Or do you have more comments?

**Comment by dg-concordium** [comment]
> @dg-concordium Can we get this deployed tomorrow? Or do you have more comments?

I can deploy now.

**Review by dg-concordium — APPROVED** [review]
